src/core/qgslogger.h

```diff
@@ -31,10 +31,12 @@ class QFile;
 
 #ifdef QGISDEBUG
 #define QgsDebugMsg(str) QgsLogger::debug(QString(str), 1, __FILE__, __FUNCTION__, __LINE__)
+#define QgsTraceMsg(str) if ( 2 <= QgsLogger::debugLevel() ) { QgsLogger::debug(QString(str), 2, __FILE__, __FUNCTION__, __LINE__); }(void)(0)
```

Comment:
Like Nyall I am also not a fan of introducing `QgsTraceMsg` - we have QgsDebugMsgLevel which is widely used in QGIS code, let's stick to using that...

Reply:
I am really not a huge fan of QgsDebugMsgLevel, I do not think, using a function QgsDebugMsgLevel with the cryptic number 2 helps to better understand the code. 

In every language I have used, there is always a logger with debug, info, warning, error levels. Sometimes even trace and critical levels (as the QMessageLogger). I would like to be able to log messages using the same design: multiple level clearly identified (by an enum).

Reply:
QgsDebugMsgLevel is used in qgis code ~300 times from a quick search, so it is fairly standard idiom for qgis developers, so introducing another idiom for the same thing may not be ideal. If we introduce a different idiom then ideally all the other QgsDebugMsgLevel occurrences should be renamed as well. I would suggest to leave these logger-related bits out of this PR and create a separate PR to introduce QgsTraceMsg where other devs may comment...

Reply:
ok, I have revert my changes and I will create a dedicated PR